include/libssh/threads.h

```diff
@@ -24,8 +24,56 @@
 #include <libssh/libssh.h>
 #include <libssh/callbacks.h>
 
+#ifdef HAVE_PTHREAD
+
+#include <pthread.h>
+#define SSH_STATIC_MUTEX(mutex) \
+    static pthread_mutex_t mutex = PTHREAD_MUTEX_INITIALIZER
+
+# define SSH_STATIC_MUTEX_LOCK(mutex) \
+    pthread_mutex_lock(&mutex)
```

Comment:
Always check return codes of pthread_mutex_lock functions! Make it a function to do that -> `ssh_mutex_lock()`

Reply:
Not addressed yet, but I agree on this, and I will address it in a new patch.